pilot/pkg/xds/delta.go

```diff
@@ -393,7 +393,6 @@ func (s *DiscoveryServer) shouldRespondDelta(con *Connection, request *discovery
 	con.proxy.WatchedResources[request.TypeUrl].NonceAcked = request.ResponseNonce
 	con.proxy.WatchedResources[request.TypeUrl].ResourceNames = currentResources
 	alwaysRespond := previousInfo.AlwaysRespond
-	previousInfo.AlwaysRespond = false
```

Comment:
Why are we removing this?

Reply:
+1, this is mirrored in ads.go as well, so if we do need to we should make sure there is a comment explaining why they are different

Reply:
it is just a cleanup. `previousInfo` is dropped after this code snippet

Reply:
I will remove that in ads.go

Reply:
Its not dropped, its a pointer to `con.proxy.WatchedResources[request.TypeUrl]` which is not replaced

Reply:
got it. My mistake,